components/feature/readerview/src/main/java/mozilla/components/feature/readerview/view/ReaderViewControlsBar.kt

```diff
@@ -0,0 +1,152 @@
+        }
+        fontGroup.check(selected)
+    }
+
+    /**
+     * Sets the font size of the current and future ReaderView sessions.
+     *
+     * Note: The readerview.js implementation under the hood scales the entire page's contents and not just
+     * the text size.
+     *
+     * @param size An integer value in the range [MIN_TEXT_SIZE] to [MAX_TEXT_SIZE].
+     */
+    override fun setFontSize(size: Int) {
+        val (incrementState, decrementState) = when {
+            size <= MIN_TEXT_SIZE -> {
+                Pair(first = true, second = false)
+            }
+            size >= MAX_TEXT_SIZE -> {
+                Pair(first = false, second = true)
+            }
+            else -> {
+                Pair(first = true, second = true)
+            }
+        }
+        fontIncrementButton.isEnabled = incrementState
+        fontDecrementButton.isEnabled = decrementState
+    }
+
+    /**
+     * Sets the color scheme of the current and future ReaderView sessions.
+     *
+     * @param scheme The applicable colour schemes available.
+     */
+    override fun setColorScheme(scheme: ColorScheme) {
+        val selected = when (scheme) {
+            ColorScheme.DARK -> R.id.mozac_feature_readerview_color_dark
+            ColorScheme.SEPIA -> R.id.mozac_feature_readerview_color_sepia
+            ColorScheme.LIGHT -> R.id.mozac_feature_readerview_color_light
+        }
+
+        colorSchemeGroup.check(selected)
+    }
+
+    /**
+     * Requests visibility and focus of the UI controls.
+     */
+    override fun showControls() {
+        visibility = View.VISIBLE
+        requestFocus()
+    }
+
+    /**
+     * Requests invisibility of the UI controls.
+     */
+    override fun hideControls() {
+        visibility = View.GONE
+    }
+
+    private fun bindViews() {
+        fontGroup = applyCheckedListener(R.id.mozac_feature_readerview_font_group) { checkedId ->
+            val fontType = when (checkedId) {
+                R.id.mozac_feature_readerview_font_sans_serif -> FontType.SANS_SERIF
+                R.id.mozac_feature_readerview_font_serif -> FontType.SERIF
+                else -> FontType.SERIF
+            }
+            listener?.onFontChanged(fontType)
+        }
+        colorSchemeGroup = applyCheckedListener(R.id.mozac_feature_readerview_color_scheme_group) { checkedId ->
+            val colorSchemeChoice = when (checkedId) {
+                R.id.mozac_feature_readerview_color_dark -> ColorScheme.DARK
+                R.id.mozac_feature_readerview_color_sepia -> ColorScheme.SEPIA
+                R.id.mozac_feature_readerview_color_light -> ColorScheme.LIGHT
+                else -> ColorScheme.DARK
+            }
+            listener?.onColorSchemeChanged(colorSchemeChoice)
+        }
+        fontIncrementButton = applyClickListener(R.id.mozac_feature_readerview_font_size_increase) {
+            listener?.onFontSizeChanged(FontSize.INCREASE)
```

Comment:
nit: don't think we really need the FontSize enum anymore. The listener could just have `onIncreaseFontSize` and `onDecreaseFontSize`.

Reply:
Ha, good point!